### DISCUSSION TITLE
How can I infer typeof SafeActionResult result?

### DISCUSSION THREAD

**Discussion by palmmymumu** [next-safe-action/next-safe-action#276] [Q&A] [answered]
I would like to create a transform function that receives a safe action function, waits for its execution, and returns data (if successful) or throws an error if there are bindArgsValidationErrors, validationErrors, or a serverError.

Currently, this is my approach to accomplish this, and everything works well; however, the result lacks type safety (Its returned type any instead).

P.S. New to TypeScript, would like to use this function with tanstack query (useQuery Fn)

<img width="595" alt="image" src="https://github.com/user-attachments/assets/bb9944a4-621c-4fcf-b733-7480fb69cca3">

**Comment by TheEdoRan** [comment]
You can infer the result of a safe action function by using the [`InferSafeActionFnResult`](https://next-safe-action.dev/docs/types/infer-types) utility type.

**Reply by palmmymumu** [reply]
As I understand it, the InferSafeActionFnResult type is { data, serverError, ...etc. }, but my goal is to only return data and its type, otherwise throw errors

**Reply by TheEdoRan** [reply, marked as answer]
To get the data type you can select the `data` prop from the result object:

```typescript
type data = InferSafeActionFnResult<typeof action>["data"];
```
